# Coding Exercises (Part 1)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code!

The correct answer is provided below your coding cell. There you can check whether your code is correct.

If you need some help or if you want to check your code, you can also have a look at the __solutions__.

### Have Fun!

--------------------------------------------------------------------------------------------------------------

## Exercise 2:  Lists and For Loops for TVM Problems with many Cashflows

Use __variables__, __lists__ and __for loops__ where appropriate!

You have the following savings plan:

- 1,000 USD today
- 1,200 USD in 1 year
- 1,500 USD in 2 years
- 2,000 USD in 3 years
- 2,500 USD in 4 years
- 3,000 USD in 5 years
- 3,500 USD in 6 years

1. __Create a list__ and __store__ the list in memory!

In [1]:
cf = [1000,1200,1500,2000,2500,3000,3500]

(result: [1000, 1200, 1500, 2000, 2500, 3000, 3500])

2. __Index__ the list for the following elements!  

Today´s saving

In [3]:
cf[0]

1000

Saving in year 3

In [5]:
cf[3]

2000

The second element of the list

In [6]:
cf[1]

1200

Final year´s saving

In [7]:
cf[-1]

3500

3. __Iterate__ over the list and __print__ each and every element.

In [9]:
for i in range(len(cf)):
  print(cf[i])

1000
1200
1500
2000
2500
3000
3500


4. Calculate the __sum__ of your savings. Use a __for loop__ and print intermediate results!

In [10]:
total_cf = 0
for i in range(len(cf)):
  total_cf += cf[i]
print(total_cf)

14700


(result: 14,700 USD)

5. Now, use two __Python built-in function__ to calculate the sum and the number of elements.

In [11]:
#sum
sum(cf)

14700

(result: 14,700 USD)

In [12]:
#lenght
len(cf)

7

(result: 7)

6. __Print__ all integers from __0 until 100__. Use __range()__!

In [13]:
for i in range(100):
  print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


7. Create and save the list n with all timestamps (0, 1, 2, 3, 4, 5, 6). Use __range()__!

In [16]:
range_6 = list(range(6+1))

8. __Reverse__ the order of n and __reassign__ n!

In [23]:
range_6 = range_6[::-1]

(result: [6, 5, 4, 3, 2, 1, 0]

In [24]:
cf

[1000, 1200, 1500, 2000, 2500, 3000, 3500]

9. Calculate the __future value__ of your account after year 6. Assume an interest rate of 4.0% p.a.

In [19]:
r = 0.04

In [28]:
fv= 0
for i in range_6:
  fv += cf[6-i]*(1+r)**(i)
  print("year:",i+1,round(fv,2))

year: 1 3500.0
year: 2 6620.0
year: 3 9324.0
year: 4 11573.73
year: 5 13328.52
year: 6 14788.5
year: 7 16053.82


(result: 16,053.82 USD)

The ABC Company evaluates to buy an additional machine that will increase future profits/cashflows by
- 200 USD in t1,
- 300 USD in t2,
- 500 USD in t3,
- 750 USD in t4,
- 1,000 USD in t5. (each cf at period´s end)<br>


10. The machine costs __2,000 USD__ (Investment in to). Calculate the Project´s __NPV__ and evaluate whether ABC should pursue the project. <br> ABC´s required rate of return (Cost of Capital) is __8%__ p.a.

In [51]:
cf = [200,300,500,750,1000]
npv = -2000
rrt = 0.08 # required_rate of return
cf

[200, 300, 500, 750, 1000]

In [52]:
for i in range(len(cf)):
  npv += cf[i]/(1+rrt)**(i+1)
  print(npv)

-1814.8148148148148
-1557.6131687242798
-1160.697048214195
-609.4246586168551
71.15853841689773


(result: 71.16 USD)

Yes since its npv is positive.

11. Would ABC accept the project with a __required rate of return of 10.0% p.a.__?

In [53]:
cf = [200,300,500,750,1000]
npv = -2000
rrt = 0.10 # required_rate of return
cf

for i in range(len(cf)):
  npv += cf[i]/(1+rrt)**(i+1)
  print(npv)


-1818.1818181818182
-1570.2479338842977
-1194.5905334335089
-682.330441909706
-61.40911885055107


No, since its npv is negative.

(result: -61.41)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code!

The correct answer is provided below your coding cell. There you can check whether your code is correct.

### Have Fun!

--------------------------------------------------------------------------------------------------------------